### ISSUE TITLE
No normalization for validation data during model evaluation

### ISSUE THREAD

**Issue by Geo99pro** [lyuwenyu/RT-DETR#312]
**Star RTDETR**
请先在RTDETR主页点击**star**以支持本项目
Star RTDETR to help more people discover this project. 

---

**Describe the bug**
Hello everyone,

First of all, I'd like to congratulate you on the work you've done on this project.

I'm in the process of adding early stop to the code. To do this, I need to calculate the validation error first.
 I've managed to calculate the validation error in the ``evaluate function`` available in ``rtdetr_pytorch/src/solver/det_engine.py`` using criterion(outputs, targets) and summing the result.

However, I noticed that the validation images used by the model are not normalized, and this from the dataloader file available in ``rtdetr_pytorch/configs/rtdetr/include/dataloader.yml``. This leads to very high validation values.

I wonder if this lack of normalization is intentional and if there's any particular reason for it on your part?

**NB:** the bbox values for the train function are indeed normalized, whereas the validation values are not, as shown in the images below.

Thanks in advance for your help. And I apologize if my question is low level, I'm learning.

![8392e721-c2f3-4b9f-b4ef-9019ce225211](https://github.com/lyuwenyu/RT-DETR/assets/112230136/9d2804c4-9335-448c-a384-a6d0319c1db2)

![b587d38a-9e75-4acc-9b10-3de2f88666b5](https://github.com/lyuwenyu/RT-DETR/assets/112230136/dd4c5d5c-c40c-47e4-a8b2-868304db3987)

**To Reproduce**
![4e2e483b-4f2d-480b-82c6-c50603cf09a2](https://github.com/lyuwenyu/RT-DETR/assets/112230136/a71d8d67-b5a8-488f-ae9a-c39a9ed31fa5)
![80072f9a-eac9-49b1-bb19-b76ed253f486](https://github.com/lyuwenyu/RT-DETR/assets/112230136/8b848b65-72bb-43ec-8505-56c4f39e8bb1)

**Comment by lyuwenyu** [comment]
Yes, You are right. We did not normalize bbox in `val_dataloader`.  You can do it by adding `L20` after `L35`

https://github.com/lyuwenyu/RT-DETR/blob/main/rtdetr_pytorch/configs/rtdetr/include/dataloader.yml#L20
https://github.com/lyuwenyu/RT-DETR/blob/main/rtdetr_pytorch/configs/rtdetr/include/dataloader.yml#L35

**Comment by Geo99pro** [comment]
Oh yes ! Of course then it can perform L131 of the function `` _transform`` in  https://github.com/lyuwenyu/RT-DETR/blob/64878acad2f58ed34579e5a5ec45da1044587e09/rtdetr_pytorch/src/data/transforms.py.

I appreciate your answer @lyuwenyu, thanks you.

**Comment by Geo99pro** [comment]
Hi Sir @lyuwenyu.

I would like to understand something that is not clear to me.
![image](https://github.com/lyuwenyu/RT-DETR/assets/112230136/baabcb8c-df78-4505-b88c-c8b4a5926a9e)

Even normalizing the validation dataset as you said, I still don't understand why the loss value is not of the same order as the train loss.

I mean for example that with an average train loss = 16.9771, I have an average val loss = 0.911.
![image](https://github.com/lyuwenyu/RT-DETR/assets/112230136/3b12f630-c18e-403c-8601-2a24b312447e)

I don't understand why.

Here's the code I use (before taking the average val loss, I divide by the len(data_loader).
![image](https://github.com/lyuwenyu/RT-DETR/assets/112230136/057859e2-5d94-4889-9035-ceac9d735b80)

Can you, please explain me if i'm doing somthing badly. Thanks

**Comment by Geo99pro** [comment]
Sir @lyuwenyu
